File: threads/src/main/java/im/threads/internal/utils/Keyboard.java
package im.threads.internal.utils;

import android.content.Context;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public final class Keyboard {

    private Keyboard() {
    }

    public static void show(final @Nullable Context context, final @Nullable View view, final long delayMills) {
        if (context != null && view != null) {
            new Handler().postDelayed(() -> show(context, view), delayMills);
        }
    }

    public static void hide(final @Nullable Context context, final @Nullable View view, final long delayMills) {
        if (context != null && view != null) {
            new Handler().postDelayed(() -> hide(context, view), delayMills);
        }
    }

    private static void show(final @Nullable Context context, final @Nullable View view) {
        if (context != null && view != null) {
            view.requestFocus();
            final InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
            if (imm != null) {
                imm.showSoftInput(view, 0);
            }
        }
    }

    private static void hide(final @Nullable Context context, final @Nullable View view) {
        if (context != null && view != null) {
            final InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
            if (imm != null) {
                imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
            }
        }
    }
}

File: src/main/java/im/threads/ui/utils/Keyboard.kt
package im.threads.ui.utils

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.view.View
import android.view.inputmethod.InputMethodManager

fun View.hideKeyboard() {
    val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    imm.hideSoftInputFromWindow(windowToken, 0)
}

fun View.hideKeyboard(delayMills: Long) {
    Handler(Looper.getMainLooper()).postDelayed({
        hideKeyboard()
    }, delayMills)
}

fun View.showKeyboard() {
    this.requestFocus()
    (context.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager)?.showSoftInput(this, 0)
}

fun View.showKeyboard(delayMills: Long) {
    Handler(Looper.getMainLooper()).postDelayed({
        showKeyboard()
    }, delayMills)
}
